lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -7177,9 +7183,7 @@ class PullbackEmitter final : public SILInstructionVisitor<PullbackEmitter> {
       for (auto i : range(ti->getNumOperands())) {
         if (!getTangentSpace(ti->getOperand(i)->getType().getASTType()))
           continue;
-        auto adjElt = val;
-        if (val->getType().is<TupleType>())
-          adjElt = elts->getResult(adjIdx++);
+        auto adjElt = elts->getResult(adjIdx++);
```

Comment:
Note: directly using `elts->getResult` is safe because `elts` is a `destructure_tuple` and `val->getType().is<TupleType>()` is always true.